sdk/nodejs/runtime/invoke.ts

```diff
@@ -213,19 +213,29 @@ export class StreamInvokeResponse<T> implements AsyncIterable<T> {
     }
 }
 
-function createInvokeRequest(tok: string, serialized: any, provider: string | undefined, opts: InvokeOptions) {
+async function createInvokeRequest(tok: string, serialized: any, provider: string | undefined, opts: InvokeOptions, packageRef?: Promise<string | undefined>) {
     if (provider !== undefined && typeof provider !== "string") {
         throw new Error("Incorrect provider type.");
     }
 
     const obj = gstruct.Struct.fromJavaScript(serialized);
-
+    let packageRefStr = undefined;
+    if (packageRef !== undefined) {
+      packageRefStr = await packageRef;
+      if (packageRefStr !== undefined) {
+        // If we have a package reference we can clear some of the resource options
+        opts.version = undefined;
+        opts.pluginDownloadURL = undefined;
+      }
+    }
     const req = new resourceproto.ResourceInvokeRequest();
     req.setTok(tok);
     req.setArgs(obj);
     req.setProvider(provider || "");
     req.setVersion(opts.version || "");
     req.setAcceptresources(!utils.disableResourceReferences);
+    req.setPlugindownloadurl(opts.pluginDownloadURL || "")
```

Comment:
We didn't set `pluginDownloadURL` before for nodejs, which seems like a bug? For param providers this is cleared above.

Reply:
This too probably worth just fixing alone. We ought to make sure we have a conformance test issue raised to cover this as well.

Reply:
Yep, removed the change here and created an issue for me to follow up.